ext/tunemygc/tunemygc_ext.c

```diff
@@ -39,26 +43,53 @@ static VALUE tunemygc_walltime(VALUE mod)
     return DBL2NUM(_tunemygc_walltime());
 }
 
+static void reverse(tunemygc_stat_record** head_ref) {
```

Comment:
To me it looks like the only reason you need this `reverse` function is to preserve the order of events. What I'm saying is that you could add a `static tunemygc_stat_record *first_cycle;` where you declare `current_cycle` that stores the head of the list. Then you could simply walk the list from `first_cycle` instead of having to reverse the linked list.